crates/sc-consensus-subspace/src/lib.rs

```diff
@@ -1182,18 +1182,16 @@ where
         self.imported_block_notification_sender
             .notify(move || (block_number, root_block_sender));
 
-        let next_block_number = block_number + One::one();
+        let mut root_blocks = Vec::new();
+
         while let Some(root_block) = root_block_receiver.next().await {
-            {
-                let mut root_blocks = self.root_blocks.lock();
-                if let Some(list) = root_blocks.get_mut(&next_block_number) {
-                    list.push(root_block);
-                } else {
-                    root_blocks.put(next_block_number, vec![root_block]);
-                }
-            }
+            root_blocks.push(root_block);
         }
 
+        self.root_blocks
+            .lock()
```

Comment:
Is it okay to use sync mutex in async function? Sometimes it can lead to exhaustion or even deadlocks

Reply:
`Mutex` usage in async code is very okay in this case, nothing holds the lock for any significant amount of time, it is always lock/unlock and move on.